compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,362 @@ module Language.Bond.Codegen.Java.Class_java
+        typeName BT_Int8 = [lt|com.microsoft.bond.StructBondType.Int8StructField|]
+        typeName (BT_Maybe BT_Int16) = [lt|com.microsoft.bond.StructBondType.SomethingInt16StructField|]
+        typeName BT_Int16 = [lt|com.microsoft.bond.StructBondType.Int16StructField|]
+        typeName (BT_Maybe BT_Int32) = [lt|com.microsoft.bond.StructBondType.SomethingInt32StructField|]
+        typeName BT_Int32 = [lt|com.microsoft.bond.StructBondType.Int32StructField|]
+        typeName (BT_Maybe BT_Int64) = [lt|com.microsoft.bond.StructBondType.SomethingInt64StructField|]
+        typeName BT_Int64 = [lt|com.microsoft.bond.StructBondType.Int64StructField|]
+        typeName (BT_Maybe BT_UInt8) = [lt|com.microsoft.bond.StructBondType.SomethingUInt8StructField|]
+        typeName BT_UInt8 = [lt|com.microsoft.bond.StructBondType.UInt8StructField|]
+        typeName (BT_Maybe BT_UInt16) = [lt|com.microsoft.bond.StructBondType.SomethingUInt16StructField|]
+        typeName BT_UInt16 = [lt|com.microsoft.bond.StructBondType.UInt16StructField|]
+        typeName (BT_Maybe BT_UInt32) = [lt|com.microsoft.bond.StructBondType.SomethingUInt32StructField|]
+        typeName BT_UInt32 = [lt|com.microsoft.bond.StructBondType.UInt32StructField|]
+        typeName (BT_Maybe BT_UInt64) = [lt|com.microsoft.bond.StructBondType.SomethingUInt64StructField|]
+        typeName BT_UInt64 = [lt|com.microsoft.bond.StructBondType.UInt64StructField|]
+        typeName (BT_Maybe BT_Float) = [lt|com.microsoft.bond.StructBondType.SomethingFloatStructField|]
+        typeName BT_Float = [lt|com.microsoft.bond.StructBondType.FloatStructField|]
+        typeName (BT_Maybe BT_Double) = [lt|com.microsoft.bond.StructBondType.SomethingDoubleStructField|]
+        typeName BT_Double = [lt|com.microsoft.bond.StructBondType.DoubleStructField|]
+        typeName (BT_Maybe BT_Bool) = [lt|com.microsoft.bond.StructBondType.SomethingBoolStructField|]
+        typeName BT_Bool = [lt|com.microsoft.bond.StructBondType.BoolStructField|]
+        typeName (BT_Maybe BT_String) = [lt|com.microsoft.bond.StructBondType.SomethingStringStructField|]
+        typeName BT_String = [lt|com.microsoft.bond.StructBondType.StringStructField|]
+        typeName (BT_Maybe BT_WString) = [lt|com.microsoft.bond.StructBondType.SomethingWStringStructField|]
+        typeName BT_WString = [lt|com.microsoft.bond.StructBondType.WStringStructField|]
+        typeName (BT_Maybe (BT_UserDefined e@Enum {..} _)) = [lt|com.microsoft.bond.StructBondType.SomethingEnumStructField<#{qualifiedDeclaredTypeName java e}>|]
+        typeName (BT_UserDefined e@Enum {..} _) = [lt|com.microsoft.bond.StructBondType.EnumStructField<#{qualifiedDeclaredTypeName java e}>|]
+        typeName (BT_Maybe t) = [lt|com.microsoft.bond.StructBondType.SomethingObjectStructField<#{(getTypeName java) t}>|]
+        typeName t = [lt|com.microsoft.bond.StructBondType.ObjectStructField<#{(getTypeName java) t}>|]
```

Comment:
It seems we're expecting the default for those types. I'm going to put this back for now.